core/src/Streamly/Internal/Data/Stream/StreamD/Nesting.hs

```diff
@@ -1400,7 +1399,7 @@ parseIterate func seed (Stream step state) =
             PRD.IDone pb ->
                 let next = ConcatParseInit [] st (func pb)
                  in return $ Skip $ ConcatParseYield pb next
-            PRD.IError err -> throwM $ ParseError err
+            PRD.IError err -> fail err
```

Comment:
We should not use `fail`. See the changes in `parseMany` we need to yield a `Left` result instead.